### ISSUE TITLE
Data balance and upsampling

### ISSUE THREAD

**Issue by cena001plus** [ultralytics/yolov5#7050]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I use voc format for training, including train.txt and val.txt, each file includes the specific path of each image, but according to the number of category boxes, I made a simple data balance for the images, such as the path of some images in The txt file is repeated many times. During training, I found that the code automatically filters the repeated data, which means that the data balance has been removed. How to solve this?

### Additional

The amount of data before data balance is 152300, and the amount of data after data balance is 269846, but it is found that the data has become 1523xx during training. According to batch_size*itera_num

**Comment by glenn-jocher** [comment]
@cena001plus the dataloader class is here, you might want to review to see what's happening:
https://github.com/ultralytics/yolov5/blob/9f4d11379bb931586c1f51c1d85c6fac9fc37eda/utils/datasets.py#L380-L381

*.txt dataset files are parsed here:
https://github.com/ultralytics/yolov5/blob/9f4d11379bb931586c1f51c1d85c6fac9fc37eda/utils/datasets.py#L404-L408

**Comment by cena001plus** [comment]
> @cena001plus the dataloader class is here, you might want to review to see what's happening:
> 
> https://github.com/ultralytics/yolov5/blob/9f4d11379bb931586c1f51c1d85c6fac9fc37eda/utils/datasets.py#L380-L381
> 
> *.txt dataset files are parsed here:
> 
> https://github.com/ultralytics/yolov5/blob/9f4d11379bb931586c1f51c1d85c6fac9fc37eda/utils/datasets.py#L404-L408

The code segment you mentioned does not affect the data balance. If the amount of data is judged according to the txt file in the generated label, this problem will occur, because multiple identical pictures will generate the same label file. This problem should not occur if the image path in the file is used to determine the training data

**Comment by cena001plus** [comment]
if .txt is follow:
`
/home/data/1.jpg
/home/data/2.jpg
/home/data/3.jpg
/home/data/3.jpg
`

.cache is dict,  it's content is : {"/home/data/1.jpg":[], "/home/data/2.jpg":[] , "/home/data/3.jpg":[]}

so this data balancing strategy is ineffective。
